cmd/kubeadm/app/cmd/init.go

```diff
@@ -461,6 +468,34 @@ func (d *initData) Tokens() []string {
 	return tokens
 }
 
+// TryLoadCertFromDisk tries to get cert from cache or disk
+func (d *initData) TryLoadCertFromDisk(pkiPath, name string) (*x509.Certificate, error) {
+	path := filepath.Join(pkiPath, name)
+	if cert, ok := d.certificates[path]; ok {
```

Comment:
This is not thread safe. We need to place a comment at the start of the func, that warns the use of this in concurrent code.

Reply:
I can make this thread safe if you think that it's necessary

Reply:
i comment would suffice for now IMHO.
our init code should be mainly single thread related to the data object.

but if e.g. parallel unit tests are ran this will break.

Reply:
Never make something thread safe if you don't run it concurrently. Unnecessary thread safety harms performance (cache misses, etc.) and, if at some point this actually begins executing concurrently, the thread safety will be sub-optimal and possibly buggy.

A comment is OK for now, just make sure this won't be called from some parallel test as @neolit123 mentioned (parallel tests are those why call `t.Parallel()` or are run via `t.Run()`).